app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import { RouteComponentProps, useParams } from "@reach/router";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Nav from "react-bootstrap/Nav";
+import Scrollspy from "react-scrollspy";
+import { AnalysisData } from "../../lib/visualization/types";
+import { analysisAvailable } from "../../lib/visualization/utils";
+import { StatusCheck } from "../../lib/experiment";
+import { DisabledItem } from "../DisabledItem";
+import { LinkNav } from "../LinkNav";
+import { ReactComponent as ChevronLeft } from "../../images/chevron-left.svg";
+import { ReactComponent as Clipboard } from "../../images/clipboard.svg";
+import { ReactComponent as BarChart } from "./bar-chart.svg";
+import LinkExternal from "../LinkExternal";
+
+export const RESULTS_LOADING_TEXT = "Checking results availability...";
+
+const getSidebarItems = (metrics: { [metric: string]: string }) => {
+  const keys: string[] = [];
+  const labels: string[] = [];
+
+  Object.keys(metrics!).map((metric) => {
+    keys.push(metric);
+    labels.push(metrics[metric]);
+  });
+
+  return { keys, labels };
+};
+
+type AppLayoutSidebarLockedProps = {
+  testid?: string;
+  children: React.ReactNode;
+  status?: StatusCheck;
+  analysis?: AnalysisData;
+  analysisLoadingInSidebar?: boolean;
+  analysisError?: Error;
+} & RouteComponentProps;
+
+export const AppLayoutSidebarLocked = ({
+  children,
+  testid = "AppLayoutSidebarLocked",
+  status,
+  analysis,
+  analysisLoadingInSidebar = false,
+  analysisError,
+}: AppLayoutSidebarLockedProps) => {
+  const { slug } = useParams();
+  const { keys, labels } = getSidebarItems(analysis.other_metrics);
```

Comment:
I'm getting a `General application error`...

```
Uncaught TypeError: can't access property "other_metrics", analysis is undefined
```

Maybe this component is trying to render before the analysis call is finished? Didn't look too deeply since this is just a quick first pass 🤔